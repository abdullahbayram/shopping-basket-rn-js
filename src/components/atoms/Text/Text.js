import * as React from 'react';
import PropTypes from 'prop-types';
import { Text as PaperText } from 'react-native-paper';

const Text = ({ children, variant = 'bodyMedium' }) => <PaperText variant={variant}>{children}</PaperText>;

Text.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
};

export default Text;
