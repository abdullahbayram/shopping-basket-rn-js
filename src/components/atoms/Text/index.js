import * as React from 'react';
import PropTypes from 'prop-types';
import { Text as PaperText, useTheme } from 'react-native-paper';

const Text = ({ children, variant = 'bodyMedium', style, numberOfLines, ellipsizeMode, testID = 'text' }) => {
  const theme = useTheme();
  return (
    <PaperText
      testID={testID}
      style={[{ color: theme.colors.tertiary }, style]}
      variant={variant}
      numberOfLines={numberOfLines}
      ellipsizeMode={ellipsizeMode}
    >
      {children}
    </PaperText>
  );
};

Text.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
  testID: PropTypes.string,
  ellipsizeMode: PropTypes.string,
  numberOfLines: PropTypes.number,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default Text;
