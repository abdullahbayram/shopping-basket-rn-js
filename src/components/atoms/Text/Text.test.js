import React from 'react';
import { render, screen } from '@testing-library/react-native';
import Text from '.'; // Adjust the import path to your file

describe('<Text />', () => {
  const testText = 'Hello, World!';
  it('matches the snapshot', () => {
    const { toJSON } = render(<Text>{testText}</Text>);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders the children correctly', () => {
    render(<Text>{testText}</Text>);
    expect(screen.getByText(testText)).toBeTruthy();
  });
  it('uses the default variant when not provided', () => {
    render(<Text>{testText}</Text>);
    const textElement = screen.getByText(testText);
    expect(textElement.props.style).toMatchSnapshot();
  });

  it('renders with the provided variant', () => {
    render(<Text variant="headlineSmall">{testText}</Text>);
    const textElement = screen.getByText(testText);
    expect(textElement.props.style).toMatchSnapshot();
  });
});
