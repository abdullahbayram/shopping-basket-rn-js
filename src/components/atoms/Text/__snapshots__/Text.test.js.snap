// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> matches the snapshot 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "writingDirection": "ltr",
      },
      [
        {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        undefined,
      ],
    ]
  }
>
  Hello, World!
</Text>
`;

exports[`<Text /> renders with the provided variant 1`] = `
[
  {
    "textAlign": "left",
  },
  {
    "color": "rgba(28, 27, 31, 1)",
    "writingDirection": "ltr",
  },
  [
    {
      "fontFamily": "System",
      "fontSize": 24,
      "fontWeight": "400",
      "letterSpacing": 0,
      "lineHeight": 32,
    },
    undefined,
  ],
]
`;

exports[`<Text /> uses the default variant when not provided 1`] = `
[
  {
    "textAlign": "left",
  },
  {
    "color": "rgba(28, 27, 31, 1)",
    "writingDirection": "ltr",
  },
  [
    {
      "fontFamily": "System",
      "fontSize": 14,
      "fontWeight": "400",
      "letterSpacing": 0.25,
      "lineHeight": 20,
    },
    undefined,
  ],
]
`;
